src/sshping.cxx

```diff
@@ -670,25 +806,60 @@ void end_session(ssh_session & ses) {
     }
 }
 
+// returns port, ip_and_port gets converted to just ip
+std::string parse_ip_address(std::string& ip_and_port) {
+    char buf[16];
+    // is ip address alone
+    if (inet_pton(AF_INET, ip_and_port.c_str(), buf) ||
+        inet_pton(AF_INET6, ip_and_port.c_str(), buf)) {
+        return "22";
+    }
+
+    // That didn't work, try splitting the port off
+
+    // Remove square brackets
+    ip_and_port.erase(remove(ip_and_port.begin(), ip_and_port.end(), '['),
+        ip_and_port.end());
+    ip_and_port.erase(remove(ip_and_port.begin(), ip_and_port.end(), ']'),
+        ip_and_port.end());
+
+    const size_t ip_and_port_colon = ip_and_port.find_last_of(':');
+    const std::string ip = ip_and_port.substr(0, ip_and_port_colon);
+    const std::string port = ip_and_port.substr(ip_and_port_colon + 1);
+    // is ip and port
+    if (inet_pton(AF_INET, ip.c_str(), buf) ||
+        inet_pton(AF_INET6, ip.c_str(), buf)) {
+        ip_and_port = ip;
+        return port;
+    }
+
+    // That didn't work check if hostname
+    if (ip_and_port_colon != std::string::npos) {
+        ip_and_port = ip;
+        return port;
+    }
+
+    // if empty string, not valid ip address
```

Comment:
This comment can be removed. I know this comment was true when we wrote the code at the start, but it's no longer true, since we had to accommodate for hostnames.